Client/game_sa/CColModelSA.h

```diff
@@ -11,77 +11,137 @@
+
+struct CColLineSA
+{
+    CVector vecStart;
+    float   startSize;
+    CVector vecStop;
+    float   stopSize;
+};
+static_assert(sizeof(CColLineSA) == 0x20, "Invalid size for CColLineSA");
 
-typedef struct
+struct CSphereSA
 {
     CVector vecCenter;
-    float   fRadius;
-} CColSphereSA;
+    float   radius;
+};
+static_assert(sizeof(CSphereSA) == 0x10, "Invalid size for CSphereSA");
 
-typedef struct
+struct CColSphereSA : CSphereSA
 {
-    CVector min;
-    CVector max;
-} CColBoxSA;
+    union
+    {
+        EColSurface  material;
+        std::uint8_t collisionSlot;
+    };
+
+    union
+    {
+        std::uint8_t flags;
 
-typedef struct
+        struct
+        {
+            std::uint8_t hasSpheresBoxesTriangles : 1;
+            std::uint8_t isSingleAllocationCollisionData : 1;
+            std::uint8_t flag0x04 : 1;
+            std::uint8_t flag0x08 : 1;
+            std::uint8_t flag0x10 : 1;
+            std::uint8_t flag0x20 : 1;
+            std::uint8_t flag0x40 : 1;
+            std::uint8_t flag0x80 : 1;
+        };
+    };
+
+    std::uint8_t lighting;
+    std::uint8_t light;
+
+    CColSphereSA()
+    {
+        collisionSlot = 0;
+        flags = 0;
+        lighting = 0;
+        light = 0;
+    }
+};
+static_assert(sizeof(CColSphereSA) == 0x14, "Invalid size for CColSphereSA");
+
+struct CColTriangleSA
 {
-    unsigned short v1;
-    unsigned short v2;
-    unsigned short v3;
-    EColSurface    material;
-    CColLighting   lighting;
-} CColTriangleSA;
-
-typedef struct
+    std::uint16_t indices[3];
+    EColSurface   material;
+    CColLighting  lighting;
+};
+static_assert(sizeof(CColTriangleSA) == 0x8, "Invalid size for CColTriangleSA");
+
+struct CColTrianglePlaneSA
 {
-    BYTE pad0[12];
-} CColTrianglePlaneSA;
+    CCompressedVectorSA normal;
+    std::uint16_t       distance;
+    std::uint8_t        orientation;
+};
+static_assert(sizeof(CColTrianglePlaneSA) == 0xA, "Invalid size for CColTrianglePlaneSA");
 
-typedef struct
+struct ColModelFileHeader
 {
     char  version[4];
     DWORD size;
     char  name[0x18];
-} ColModelFileHeader;
+};
 
-typedef struct
+struct CColDataSA
 {
-    WORD                 numColSpheres;
-    WORD                 numColBoxes;
-    WORD                 numColTriangles;
-    BYTE                 ucNumWheels;
-    BYTE                 pad3;
-    CColSphereSA*        pColSpheres;
-    CColBoxSA*           pColBoxes;
-    void*                pSuspensionLines;
-    void*                pUnknown;
-    CColTriangleSA*      pColTriangles;
-    CColTrianglePlaneSA* pColTrianglePlanes;
-} CColDataSA;
-
-class CColModelSAInterface
+    std::uint16_t        numSpheres;
+    std::uint16_t        numBoxes;
+    std::uint16_t        numTriangles;
+    std::uint8_t         numSuspensionLines;
+    std::uint8_t         flags;
+    CColSphereSA*        spheres;
+    CColBoxSA*           boxes;
+    CColLineSA*          suspensionLines;
```

Comment:
In theory its like this:
```suggestion
    union {
        CColLine* lines;
        CColDisk* disks;
    };
```